src/Tests/Public/ResourceProviders/Microsoft.Network/DnsZone/New-ArmPrivateDnsZoneVirtualNetworkLink.Tests.ps1

```diff
@@ -0,0 +1,89 @@
+                        PSTypeName  = "PDNSZVirtualNetworkLink"
+                        type        = $ResourceType
+                        name        = "[concat('$ExpectedPrivateDnsZoneName/$ExpectedName-$ExpectedResourceGroupName-', $ExpectedVirtualNetworkName)]"
+                        location    = "global"
+                        apiVersion  = "2018-09-01"
+                        properties  = @{
+                            registrationEnabled = $false
+                            virtualNetwork = @{
+                                id = $VirtualNetwork._ResourceId
+                            }
+                        }
+                        dependsOn   = @()
+                    }
+                    Types = $expectedTypes
+                }
+            ) {
+                param($Name, $PrivateDnsZone, $VirtualNetwork, $ResourceGroupName, $Expected, $Types)
+
+                $Expected.PSTypeNames.Add("ArmResource")
+
+                $actual = $Name | New-ArmPrivateDnsZoneVirtualNetworkLink `
+                    -ResourceGroupName $ResourceGroupName `
+                    -VirtualNetwork $VirtualNetwork `
+                    -PrivateDnsZone $PrivateDnsZone
+
+                ($actual | ConvertTo-Json -Depth $Depth -Compress | ForEach-Object { [System.Text.RegularExpressions.Regex]::Unescape($_) }) `
+                    | Should -BeExactly ($Expected | ConvertTo-Json -Depth $Depth -Compress| ForEach-Object { [System.Text.RegularExpressions.Regex]::Unescape($_) })
+
+                $Types | ForEach-Object { $actual.PSTypeNames | Should -Contain $_ }
+            }
+        }
+
+        Context "Integration tests" {
+            It "Default" -Test {
+                Invoke-IntegrationTest -ArmResourcesScriptBlock `
+                {
+                    $ExpectedName | New-ArmPrivateDnsZoneVirtualNetworkLink `
+                        -ResourceGroupName $ExpectedResourceGroupName `
+                        -VirtualNetwork $VirtualNetwork `
+                        -PrivateDnsZone $PrivateDnsZone `
+                        | Add-ArmResource
+                }
+            }
+
+            It "Multiple"  -Test {
+                Invoke-IntegrationTest -ArmResourcesScriptBlock `
+                {
+                    $PrivateDnsZoneVirtualNetworkLinks = @()
+                    for ($i = 0; $i -lt 5; $i++) {
+                        $PrivateDnsZoneVirtualNetworkLinks += @(
+                            "$ExpectedName$i" | New-ArmPrivateDnsZoneVirtualNetworkLink `
+                                -ResourceGroupName $ExpectedResourceGroupName `
+                                -VirtualNetwork $VirtualNetwork `
+                                -PrivateDnsZone $PrivateDnsZone `
+                        )
+                    }
+                    $PrivateDnsZoneVirtualNetworkLinks | Add-ArmResource
+                }
+            }
+        }
+    }
```

Comment:
Doesn't the multiple case also implicitly cover the single case?

Reply:
I think the single case covers the multiple case as the resources are added as separate elements, so I will remove this one

Reply:
I think the single case covers the multiple case as the resources are added as separate elements, so I will remove this one